internal/service/wafv2/web_acl_data_source.go

```diff
@@ -88,3 +103,44 @@ func dataSourceWebACLRead(ctx context.Context, d *schema.ResourceData, meta inte
 
 	return diags
 }
+
+func getWebAclDataSourceByArn(ctx context.Context, conn *wafv2.WAFV2, arn string, d *schema.ResourceData, diags diag.Diagnostics) diag.Diagnostics {
+	input := &wafv2.GetWebACLForResourceInput{
+		ResourceArn: aws.String(arn),
+	}
+
+	output, err := conn.GetWebACLForResourceWithContext(ctx, input)
+	if err != nil {
+		return sdkdiag.AppendErrorf(diags, "reading WAFv2 WebACLs: %s", err)
+	}
+
+	if output == nil || output.WebACL == nil {
+		return sdkdiag.AppendErrorf(diags, "WAFv2 WebACL not found for source: %s", arn)
```

Comment:
Should we instead just return nothing, if there is no attachment instead of  throwing an error?

Reply:
Went with the existing functionality, if the scope and name aren't there it will also throw an error

Reply:
Got it, that totally makes sense when the `name` is provided, because then we’re specifically searching for a certain ACL. For `resource`, it might be nicer from a user’s perspective to see an empty value rather than an error if nothing is attached. Also, when we use FMS, attachments don't happen immediately. So, if the same code is tagging the resource and then immediately trying to use data from that source, it could run into issues due to the delay with FMS.

Reply:
The name call is actually less direct than using resource arn. It lists everything by scope then iterates through to filter out the name, this call gets the WAF explicitly by resource_arn input. I think in general if a data source isn't returned you want an error to be thrown at the source, otherwise it will just get thrown later when you try to use an empty datasource attribute.

Reply:
Using empty results data sources is expected and can be worked around on terraform level, but if the data source itself fails it blocks the terraform flow completely and will require targeted deployment.